src/Raven.Server/Documents/AI/AiIntegrationsController.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Embeddings;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.ETL.Providers.AI.Embeddings;
+using Raven.Server.Documents.ETL.Providers.AI;
+using Raven.Client.Documents.Operations.AI;
+using Raven.Client.Util;
+using Raven.Server.Documents.AI.Embeddings;
+using Sparrow.Server.Logging;
+
+#pragma warning disable SKEXP0001
+
+namespace Raven.Server.Documents.AI;
+
+public class AiIntegrationsController : IDisposable
+{
+    private readonly Dictionary<AiConnectionStringIdentifier, (AiConnectionString, ITextEmbeddingGenerationService)> _embeddingsGenerationServiceByConnectionStringIdentifier;
```

Comment:
The entire class _has_ to be thread safe, but it uses things like `Dictionary<>`, which even at a glance doesn't seem like it would work for that purpose.

Reply:
It was using dictonaries only for reads (and updates replaced whole instances). But this got changed in the recent commits so now it needs to be fixed.